### PR TITLE
Integrate bamboo to improve working hours reports

### PR DESCRIPTION
**Pull request by mirland**
# Description
- The new report calculates how many hours the user has to work and it compares it with the worked hours

### CODE REVIEW

File: src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                logger.info("Work hours report from $from to $to generated in $calculationTime)")
+            }
+    }
+
+    private suspend fun getBambooReport(from: LocalDate, to: LocalDate): List<BambooWorkTimeReport> = coroutineScope {
+        require(!from.isAfter(to))
+        (async { BambooHrReportsRemoteSource.getUsers() } to
+                async { BambooHrReportsRemoteSource.getTimeOff(from, to) })
+            .let { it.first.await() to it.second.await() }
+            .let { (users, timeOffs) ->
```
Comment (f7deleon) [inline src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt:142]: What about something like, and removing the let? 

```suggestion
        val (users, timeOffs) = (async { BambooHrReportsRemoteSource.getUsers() }.await() to
                async { BambooHrReportsRemoteSource.getTimeOff(from, to).await() })
```

**Review by f7deleon — APPROVED** [review]

File: src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt (hunk 1 above)
Comment (f7deleon) [reply src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt:142]: and you map the generated collection?

Comment (mirland) [reply src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt:142]: It's not the same right, in that way getUsers will be executed before getTimeOff right?

Comment (mirland) [reply src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt:142]: Yes I tested, 
```
    GlobalScope.launch {
        val start = Date()
        val (value1, value2) = async { delay(1000); 1 }.await() to async { delay(1000); 1 }.await()
        println("*++++++++++" + (Date().time - start.time))
    }
```
the output is 2010

However if I use: 
```
    GlobalScope.launch {
        val start = Date()
        val (value1, value2) = (async { delay(1000); 1 } to async { delay(1000); 1 })
            .let { it.first.await() to it.second.await() }
        println("*++++++++++" + (Date().time - start.time))
    }
```
The output is 1008

Comment (f7deleon) [reply src/com/xmartlabs/slackbot/manager/WorkTimeReportManager.kt:142]: what about using variables? instead f let?

**Review by f7deleon — APPROVED** [review]

**Pull request merged**